src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -174,14 +171,14 @@ public void RemoveAt(int index)
         {
             get
             {
-                if (index >= _size)
+                if (index < 0 || index >= _size)
```

Comment:
If index < 0 exception ArgumentOutOfRange should be raised. You suggests strange code.